### PR TITLE
Add ColorPickerModal component (with issues fixed) to custom-components

### PR DESCRIPTION
**Pull request by beebls**
Due to some issues with presumably the fact that I am not using WSL, I was unable to commit or rebase my previous fork as I would get error messages. I had to create a new fork of the repo that was up to date, but this is the same component as before.
I've fixed the global variable concerns raised previously, but I do implore you to look through it to ensure no more possible conflicts have been left in.

This was originally designed for the CSSLoader plugin to choose custom colours, but I modified it to be universal and expose all the necessary variables.

It must be wrapped in a showModal() as it inherits the closeModal property of that.
The other properties are: 

- `onConfirm(HSLString: string, closeModal: () => void): any;`, this is a function the user specifies that is given 2 parameters, the first being an HSL color string, and the second being the closeModal function.
- `title: string;`, this is the title that is shown in the Modal window.
- `defaultH: number;`, the starting hue value
- `defaultS: number;`, the starting saturation value
- `defaultL: number;`, the starting lightness value

All of these are in the interface for the component.

I was inspired to PR this because AAGaming asked me to add it, but I don't know too much about the inner workings of the decky-frontend-lib. I built the dist folder and everything worked perfectly, but if someone else could ensure it builds properly as well it would be greatly appreciated 😄 

Image

### CODE REVIEW

File: src/custom-components/ColorPickerModal.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  closeModal: () => void;
+  onConfirm?(HSLString: string): any;
+  title?: string;
+  defaultH?: number;
+  defaultS?: number;
+  defaultL?: number;
+  defaultA?: number;
+}
+
+export const ColorPickerModal: VFC<ColorPickerModalProps> = ({
```
Comment (AAGaming00) [inline src/custom-components/ColorPickerModal.tsx:14]: IIRC this should be an FC, not a VFC

File: src/deck-components/static-classes.ts [hunk 2]
```diff
@@ -307,9 +307,60 @@ type PlaySectionClasses = Record<
...
+
 export const staticClasses: StaticClasses = findModule((mod) => typeof mod === 'object' && mod.TransitionMenuDelay);
 export const scrollClasses: ScrollClasses = findModule((mod) => typeof mod === 'object' && mod.ScrollPanel && mod.ScrollY);
 export const gamepadDialogClasses: GamepadDialogClasses = findModule((mod) => typeof mod === 'object' && mod.WithFirstRow);
 export const quickAccessControlsClasses: QuickAccessControlsClasses = findModule((mod) => typeof mod === 'object' && mod.PanelSectionRow);
 export const updaterFieldClasses: UpdaterFieldClasses = findModule((mod) => typeof mod === 'object' && mod.PatchNotes && mod.PostedTime);
 export const playSectionClasses: PlaySectionClasses = findModule((mod) => typeof mod === 'object' && mod.MenuButton && mod.MenuActive);
+export const gamepadSliderClasses: GamepadSliderClasses = findModule((mod) => typeof mod === 'object' && mod.SliderTrack && mod.SliderHasNotches);
+
+
```
Comment (AAGaming00) [inline src/deck-components/static-classes.ts:366]: extra spacing?

**Review by AAGaming00 — CHANGES_REQUESTED** [review]

File: src/custom-components/ColorPickerModal.tsx (hunk 1 above)
Comment (beebls) [reply src/custom-components/ColorPickerModal.tsx]: I have addressed and fixed this in my newest commit to the repo.
To be completely honest, I barely know the difference and use them interchangeably, so thank you for pointing it out.

File: src/deck-components/static-classes.ts (hunk 2 above)
Comment (beebls) [reply src/deck-components/static-classes.ts]: Fixed in newest commit.

**Review by AAGaming00 — APPROVED** [review]
LGTM

**Pull request merged**